axon-server-connector/src/main/java/org/axonframework/axonserver/connector/query/AxonServerQueryBus.java

```diff
@@ -249,8 +250,28 @@ public <R> Registration subscribe(@Nonnull String queryName,
         return new AxonServerRegistration(localRegistration, serverRegistration::cancel);
     }
 
+    private <Q, R> CompletableFuture<QueryResponseMessage<R>> tryToRunQueryLocally(
+            @Nonnull QueryMessage<Q, R> queryMessage) {
+
+        if (localSegment instanceof SimpleQueryBus) {
+            SimpleQueryBus qb = (SimpleQueryBus) localSegment;
+            if (qb.hasHandlersForMessage(queryMessage)) {
+                return localSegment.query(queryMessage);
+            }
+        }
+
+        return null;
+    }
+
     @Override
     public <Q, R> CompletableFuture<QueryResponseMessage<R>> query(@Nonnull QueryMessage<Q, R> queryMessage) {
+
```

Comment:
Some tests for these changes would be necessary to proceed I think.

Reply:
of course, I will add all tests when we agree on final version of the code